types.go

```diff
@@ -248,6 +248,21 @@ type CertificateChain struct {
 	Entries []ASN1Cert `tls:"minlen:0,maxlen:16777215"`
 }
 
+// PrecertChainEntryHash is an extended PrecertChainEntry type with the
+// IssuanceChainHash field added to store the hash of the
+// CertificateChain field of PrecertChainEntry.
+type PrecertChainEntryHash struct {
+	PreCertificate    ASN1Cert `tls:"minlen:1,maxlen:16777215"`
+	IssuanceChainHash []byte   `tls:"minlen:0,maxlen:256"`
```

Comment:
maxlen should probably be 255? Same below.

Reply:
There's a lot of tls marshalling going on already across ctfe (e.g. `ExtraData` gets tls marshalled before being sent to `LogServer`, and CT structs use it too), so probably ok to use it here too for uniformity.

Reply:
I believe 256 is correct as the maximum length although we currently do not have a hash size of 256 bytes algorithm. Shouldn't the length be a multiple of 2?

For the encoding, this is part of the `ExtraData` field which is marshalled by the `tls.Marshal`.
https://github.com/google/certificate-transparency-go/blob/c77f036ca4765feb5edc4aa50e803aa5ca427b0f/trillian/util/log_leaf.go#L72

Reply:
I think the maxlen is inclusive, so as specified this allows a length of 257 because it starts at 0. The entry above this has a max length of `2^24 - 1`, which seems at odds with this one being set to `2^8 - 0`

Reply:
The test shows the `maxlen` needs to be at least equal to the slice length, that's why I believe 256 is correct in this case.

https://github.com/google/certificate-transparency-go/blob/7382f9baf5c38c36b33b8aa3c882cee45d0a9f9e/tls/tls_test.go#L330

Reply:
Cool. I still don't understand why the others are set to 2^N-1, but not worth chasing down.